### ISSUE TITLE
Unable to install docker on RHEL 7

### ISSUE THREAD

**Issue by kkorada** [docker/docker#22656]
---------------------------------------------------
UNABLE TO INSTALL DOCKER ON RHEL 7
---------------------------------------------------

kernel version: 3.10.0-123.el7.x86_64

OS: RHEL 7 Mapio

OUTPUT OF **sudo yum install docker-engine**

Resolving Dependencies
--> Running transaction check
---> Package docker-engine.x86_64 0:1.11.1-1.el7.centos will be installed
--> Processing Dependency: docker-engine-selinux >= 1.11.1-1.el7.centos for package: docker-engine-1.11.1-1.el7.centos.x86_64
--> Processing Dependency: selinux-policy >= 3.13.1-23 for package: docker-engine-1.11.1-1.el7.centos.x86_64
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_97)(64bit) for package: docker-engine-1.11.1-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0(LIBSYSTEMD_209)(64bit) for package: docker-engine-1.11.1-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0()(64bit) for package: docker-engine-1.11.1-1.el7.centos.x86_64
--> Running transaction check
---> Package docker-engine.x86_64 0:1.11.1-1.el7.centos will be installed
--> Processing Dependency: selinux-policy >= 3.13.1-23 for package: docker-engine-1.11.1-1.el7.centos.x86_64
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_97)(64bit) for package: docker-engine-1.11.1-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0(LIBSYSTEMD_209)(64bit) for package: docker-engine-1.11.1-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0()(64bit) for package: docker-engine-1.11.1-1.el7.centos.x86_64
---> Package docker-engine-selinux.noarch 0:1.11.1-1.el7.centos will be installed
--> Processing Dependency: selinux-policy-base >= 3.13.1-23 for package: docker-engine-selinux-1.11.1-1.el7.centos.noarch
--> Processing Dependency: selinux-policy-targeted >= 3.13.1-23 for package: docker-engine-selinux-1.11.1-1.el7.centos.noarch
--> Processing Dependency: policycoreutils-python for package: docker-engine-selinux-1.11.1-1.el7.centos.noarch
--> Running transaction check
---> Package docker-engine.x86_64 0:1.11.1-1.el7.centos will be installed
--> Processing Dependency: selinux-policy >= 3.13.1-23 for package: docker-engine-1.11.1-1.el7.centos.x86_64
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_97)(64bit) for package: docker-engine-1.11.1-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0(LIBSYSTEMD_209)(64bit) for package: docker-engine-1.11.1-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0()(64bit) for package: docker-engine-1.11.1-1.el7.centos.x86_64
---> Package docker-engine-selinux.noarch 0:1.11.1-1.el7.centos will be installed
--> Processing Dependency: selinux-policy-base >= 3.13.1-23 for package: docker-engine-selinux-1.11.1-1.el7.centos.noarch
--> Processing Dependency: selinux-policy-targeted >= 3.13.1-23 for package: docker-engine-selinux-1.11.1-1.el7.centos.noarch
---> Package policycoreutils-python.x86_64 0:2.2.5-11.el7 will be installed
--> Processing Dependency: libsemanage-python >= 2.1.10-1 for package: policycoreutils-python-2.2.5-11.el7.x86_64
--> Processing Dependency: audit-libs-python >= 2.1.3-4 for package: policycoreutils-python-2.2.5-11.el7.x86_64
--> Processing Dependency: python-IPy for package: policycoreutils-python-2.2.5-11.el7.x86_64
--> Processing Dependency: libqpol.so.1(VERS_1.4)(64bit) for package: policycoreutils-python-2.2.5-11.el7.x86_64
--> Processing Dependency: libqpol.so.1(VERS_1.2)(64bit) for package: policycoreutils-python-2.2.5-11.el7.x86_64
--> Processing Dependency: libapol.so.4(VERS_4.0)(64bit) for package: policycoreutils-python-2.2.5-11.el7.x86_64
--> Processing Dependency: checkpolicy for package: policycoreutils-python-2.2.5-11.el7.x86_64
--> Processing Dependency: libqpol.so.1()(64bit) for package: policycoreutils-python-2.2.5-11.el7.x86_64
--> Processing Dependency: libapol.so.4()(64bit) for package: policycoreutils-python-2.2.5-11.el7.x86_64
--> Running transaction check
---> Package audit-libs-python.x86_64 0:2.3.3-4.el7 will be installed
---> Package checkpolicy.x86_64 0:2.1.12-6.el7 will be installed
---> Package docker-engine.x86_64 0:1.11.1-1.el7.centos will be installed
--> Processing Dependency: selinux-policy >= 3.13.1-23 for package: docker-engine-1.11.1-1.el7.centos.x86_64
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_97)(64bit) for package: docker-engine-1.11.1-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0(LIBSYSTEMD_209)(64bit) for package: docker-engine-1.11.1-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0()(64bit) for package: docker-engine-1.11.1-1.el7.centos.x86_64
---> Package docker-engine-selinux.noarch 0:1.11.1-1.el7.centos will be installed
--> Processing Dependency: selinux-policy-base >= 3.13.1-23 for package: docker-engine-selinux-1.11.1-1.el7.centos.noarch
--> Processing Dependency: selinux-policy-targeted >= 3.13.1-23 for package: docker-engine-selinux-1.11.1-1.el7.centos.noarch
---> Package libsemanage-python.x86_64 0:2.1.10-16.el7 will be installed
---> Package python-IPy.noarch 0:0.75-6.el7 will be installed
---> Package setools-libs.x86_64 0:3.3.7-46.el7 will be installed
--> Finished Dependency Resolution
Error: Package: docker-engine-1.11.1-1.el7.centos.x86_64 (dockerrepo)
           Requires: libdevmapper.so.1.02(DM_1_02_97)(64bit)
Error: Package: docker-engine-1.11.1-1.el7.centos.x86_64 (dockerrepo)
           Requires: libsystemd.so.0()(64bit)
Error: Package: docker-engine-1.11.1-1.el7.centos.x86_64 (dockerrepo)
           Requires: libsystemd.so.0(LIBSYSTEMD_209)(64bit)
Error: Package: docker-engine-selinux-1.11.1-1.el7.centos.noarch (dockerrepo)
           Requires: selinux-policy-base >= 3.13.1-23
           Installed: selinux-policy-targeted-3.12.1-153.el7.noarch (@anaconda/7.0)
               selinux-policy-base = 3.12.1-153.el7
           Available: selinux-policy-minimum-3.12.1-153.el7.noarch (MYREPO)
               selinux-policy-base = 3.12.1-153.el7
           Available: selinux-policy-mls-3.12.1-153.el7.noarch (MYREPO)
               selinux-policy-base = 3.12.1-153.el7
Error: Package: docker-engine-1.11.1-1.el7.centos.x86_64 (dockerrepo)
           Requires: selinux-policy >= 3.13.1-23
           Installed: selinux-policy-3.12.1-153.el7.noarch (@anaconda/7.0)
               selinux-policy = 3.12.1-153.el7
Error: Package: docker-engine-selinux-1.11.1-1.el7.centos.noarch (dockerrepo)
           Requires: selinux-policy-targeted >= 3.13.1-23
           Installed: selinux-policy-targeted-3.12.1-153.el7.noarch (@anaconda/7.0)
               selinux-policy-targeted = 3.12.1-153.el7
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

**Comment by fntlnz** [comment]
Can you try with

```
curl -sSL https://get.docker.com/ | sh
```

**Comment by runcom** [comment]
Please follow https://docs.docker.com/engine/installation/linux/rhel/
we're not new to that kind of error though. Did you yum update & yum upgrade your box? I can install it just fine on my rhel72

**Comment by runcom** [comment]
could you try enabling server-extras-rpms? `subscription-manager repos --enable rhel-7-server-extras-rpms`

**Comment by runcom** [comment]
might also be related to https://github.com/docker/docker/issues/20484 ping @thaJeztah

**Comment by thaJeztah** [comment]
Yes, I suspect it's either the machine not being fully updated, or a missing subscription

**Comment by kkorada** [comment]
@runcom and  @thaJeztah you are right i dont have a subscription. Is there a workaround for it ?

**Comment by kkorada** [comment]
while running `-sSL https://get.docker.com/ | sh` i got below error

curl: (60) Peer's Certificate issuer is not recognized.
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.

**Comment by kkorada** [comment]
while enabling `subscription-manager repos --enable rhel-7-server-extras-rpms` i got following output
**Error: rhel-7-server-extras-rpms is not a valid repo ID. Use --list option to see valid repos.**

**Comment by ArthurChiao** [comment]
I got the same error when following the [official installation guide](https://docs.docker.com/engine/installation/linux/centos/).

Currently i'm using a older version of docker for workaround, as mentioned in [#20484](https://github.com/docker/docker/issues/20484):
```shell
$ sudo yum install -y http://yum.dockerproject.org/repo/main/centos/7/Packages/docker-engine-1.9.1-1.el7.centos.x86_64.rpm
```

my system info:
```shell
$ cat /etc/redhat-release
CentOS Linux release 7.1.1503 (Core)
$ cat /etc/centos-release
CentOS Linux release 7.1.1503 (Core)

```

**Comment by thaJeztah** [comment]
@ArthurChiao did you fully upgrade your system? Looks like you're at CentOS 7.1, whereas 7.2 is the current version.

**Comment by ArthurChiao** [comment]
@thaJeztah no,i haven't upgraded my system, and later i found that the docker i installed still has problems, such as, `sudo docker info` and `sudo docker images` are ok, but `sudo docker run <some images>` not works as expected.

**Comment by thaJeztah** [comment]
the ‘sudo docker run` sounds like a different issue, so feel free to open a new one if needed

**Comment by edenzik** [comment]
My horrible hack is to install the RPM directly and hope Docker can use the new gblic version:
```
wget https://yum.dockerproject.org/repo/main/centos/7/Packages/docker-engine-1.10.0-1.el7.centos.x86_64.rpm

sudo rpm -i --nodeps docker-engine-1.10.0-1.el7.centos.x86_64.rpm
```

**Comment by thaJeztah** [comment]
closing because there's various issues mixed in this issue. feel free to open a new issue if you're having trouble installing Docker on RHEL

**Comment by vikaschenny** [comment]
am trying to install docker-ee 17.3.0 stable version in redhat 7.2 but i got following error . can any one help me



[root@dockerslaveone yum.repos.d]# yum install docker-ee
Loaded plugins: langpacks, product-id, search-disabled-repos, subscription-manager
This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.
docker-ee-stable-17.03                                                                                                                                                   | 2.9 kB  00:00:00
Resolving Dependencies
--> Running transaction check
---> Package docker-ee.x86_64 0:17.03.2.ee.4-1.el7.centos will be installed
--> Processing Dependency: docker-ee-selinux >= 17.03.2.ee.4-1.el7.centos for package: docker-ee-17.03.2.ee.4-1.el7.centos.x86_64
--> Processing Dependency: libltdl.so.7()(64bit) for package: docker-ee-17.03.2.ee.4-1.el7.centos.x86_64
--> Processing Dependency: libseccomp.so.2()(64bit) for package: docker-ee-17.03.2.ee.4-1.el7.centos.x86_64
--> Running transaction check
---> Package docker-ee.x86_64 0:17.03.2.ee.4-1.el7.centos will be installed
--> Processing Dependency: libltdl.so.7()(64bit) for package: docker-ee-17.03.2.ee.4-1.el7.centos.x86_64
--> Processing Dependency: libseccomp.so.2()(64bit) for package: docker-ee-17.03.2.ee.4-1.el7.centos.x86_64
---> Package docker-ee-selinux.noarch 0:17.03.2.ee.4-1.el7.centos will be installed
--> Processing Dependency: policycoreutils-python for package: docker-ee-selinux-17.03.2.ee.4-1.el7.centos.noarch
--> Finished Dependency Resolution
Error: Package: docker-ee-selinux-17.03.2.ee.4-1.el7.centos.noarch (docker-ee-stable-17.03)
           Requires: policycoreutils-python
Error: Package: docker-ee-17.03.2.ee.4-1.el7.centos.x86_64 (docker-ee-stable-17.03)
           Requires: libseccomp.so.2()(64bit)
Error: Package: docker-ee-17.03.2.ee.4-1.el7.centos.x86_64 (docker-ee-stable-17.03)
           Requires: libltdl.so.7()(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
[root@dockerslaveone yum.repos.d]#

**Comment by thaJeztah** [comment]
@vikaschenny you don't have an active Red Hat subscription, so I suspect because of that you cannot install the dependencies required for Docker;

```
This system is not registered to Red Hat Subscription Management.
You can use subscription-manager to register.
```

**Comment by vikaschenny** [comment]
ya but i did mount another rhel7.iso file loot at once mounted  steps  -> https://access.redhat.com/solutions/1355683
after that i don't have any dependency error but i got another error   

failure: repodata/repomd.xml from docker-ee-stable-17.03: [Errno 256] No more mirrors to try.
https://storebits.docker.com/ee/rhel/sub-1ed280da-56b8-4254-beaa-52aa515de231/7.3/x86_64/stable-17.03/repodata/repomd.xml: [Errno 14] curl#6 - "Could not resolve host: storebits.docker.com; Name or service not known"
